common/moduleconfig/moduleconfig.go

```diff
@@ -14,12 +15,21 @@ type ModuleGoConfig struct{}
 // ModuleKotlinConfig is language-specific configuration for Kotlin modules.
 type ModuleKotlinConfig struct{}
 
+// ModuleType represents the type of the module.
+type ModuleType int
+
+const (
+	FTLModule ModuleType = iota
+	ExternalLibrary
```

Comment:
What do we think of "external library" as what we call these things?
Trying to use a name separate from ExternalModule since that's already used for imported modules

Reply:
I don't think this should be part of the ModuleConfig - it is designed only for FTL modules, so it doesn't make sense for it to be used for non-FTL code. If you're trying to reuse the ModuleConfig type somewhere, a better approach would be to create a new intermediate type that is usable in both situations and translate the ModuleConfig to that when necessary.

Reply:
Yeh, it doesn't really make sense here. Will separate it out 👍